src/main/java/gregtech/common/items/behaviors/DataItemBehavior.java

```diff
@@ -41,7 +42,7 @@ public void addInformation(@NotNull ItemStack itemStack, List<String> lines) {
                 .getDataStickEntry(researchId);
         if (recipes != null && !recipes.isEmpty()) {
             lines.add(I18n.format("behavior.data_item.assemblyline.title"));
-            Collection<ItemStack> added = new ObjectOpenHashSet<>();
+            Collection<ItemStack> added = new ObjectOpenCustomHashSet<>(ItemStackHashStrategy.comparingAllButCount());
```

Comment:
Do we want to use comparingAllButCount here? What if a recipe is added as a sort of late game alternative recipe, outputting the same item as an earlier recipe, but just more of it. Would that get us back into the situation where two item names are displayed again, or one recipe is not displayed at all.

Might just want to use `comparingAll`.